backend/src/main/java/com/huellapositiva/application/exception/BadProposalStatusException.java

```diff
@@ -0,0 +1,11 @@
+package com.huellapositiva.application.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.BAD_REQUEST)
```

Comment:
BAD_REQUEST 400 is only for malformed payload. Use 409 instead.
The name is misleading too, ProposalNotPublishableException would be more accurate.

When a request is malformed it does not matter how many times you repeat that request that you will ALWAYS get a 400 response.

This is not the case for a request that was perfectly valid but returned 409 due to the internal state of the proposal.

SCENARIO 1
-> Malformed/not-valid  request returns 400
...
-> at some point in time the proposal becomes REVIEW_PENDING
...
-> repeating the same exact malformed/not-valid request still returns 400 (forever)

SCENARIO 2
-> valid request returns 409 due to the internal state of the proposal not being publishable
...
-> at some point in time the proposal becomes REVIEW_PENDING
-> repeating same exact valid request returns 200


See the difference between a malformed/invalid request and a valid request that does not succeed because of our internal state?

Reply:
Thank you very much for taking your time in explaining the difference. I think I have a better understanding now. Noted and changed!